distributed/worker.py

```diff
@@ -4183,8 +4251,6 @@ def validate_task_fetch(self, ts):
         assert ts.key not in self.data
         assert self.address not in ts.who_has
         assert not ts.done
-        assert ts in self.data_needed
-        assert ts.who_has
```

Comment:
As motivated in the description, we rely now on the fact that there are tasks in fetch with this being empty but we rely on the transition system to pick this up and transition the task. This allows us to write easier recommendations

Reply:
`assert ts in self.data_needed` is still valid, as the validation only happens at the end of a transitions cycle. It's only in `transition_fetch_missing`, which is in the middle of a cycle, that it is not.